### ISSUE TITLE
Why are chunks required in FactualConsistencyMetric

### ISSUE THREAD

**Issue by FloTeu** [confident-ai/deepeval#84]
Hey guys,

I just started today exploring you great library and was curious to understand the factual consistency metric. 

Maybe i didnt got it right, but why do we have to create chunks of our context? It seems like they have no impact at all, since 
```python 
scores = self.model.predict([(context, output), (output, context)])
```
is always called with context and output, hence, producing the same scoring results in each loop. The `max_score` can be found in the first loop iteration allready. 

Code: deepeval/metrics/factual_consistency.py:19-32
```python
def measure(self, output: str, context: str):
    context_list = chunk_text(context)
    max_score = 0
    for c in context_list:
        scores = self.model.predict([(context, output), (output, context)])
        print(scores)
        # https://huggingface.co/cross-encoder/nli-deberta-base
        # label_mapping = ["contradiction", "entailment", "neutral"]
        softmax_scores = softmax(scores)
        score = softmax_scores[0][1]
        if score > max_score:
            max_score = score

        second_score = softmax_scores[1][1]
        if second_score > max_score:
            max_score = second_score
```

**Comment by ColabDog** [comment]
Good question. 
With NLI models, we chunk it because the score of the NLI models decrease as the length of the text decreases. 
This is a known phenomenon that can be seen in this paper. https://arxiv.org/pdf/2111.09525.pdf 

The loop in the `context_list` , however is a bug. The `context` needs to be `c`. Thanks for reporting that - will push a fix soon. 

Note to self: Also make sure to reference that paper both in the documentation and implementation.

**Comment by ColabDog** [comment]
We've added the reference paper to the documentation and fixed it - released with version 0.12.4 :)
